core/src/test/java/org/elasticsearch/action/bulk/TransportShardBulkActionTests.java

```diff
@@ -503,6 +506,53 @@ public void testCalculateTranslogLocation() throws Exception {
 
     }
 
+    public void testMappingUpdateDoesNotParseDocTwice() throws Exception {
```

Comment:
can we make sure IndexShard.prepareIndexOnPrimary is only call twice? I thought that was the purpose of this test?